subprojects/docs/src/docs/userguide/ideaPlugin.adoc

```diff
@@ -0,0 +1,242 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[idea_plugin]]
+== The IDEA Plugin
+
+The IDEA plugin generates files that are used by http://www.jetbrains.com/idea/[IntelliJ IDEA], thus making it possible to open the project from IDEA (`File` - `Open Project`). Both external dependencies (including associated source and Javadoc files) and project dependencies are considered.
+
+What exactly the IDEA plugin generates depends on which other plugins are used:
+
+.IDEA plugin behavior
+[cols="a,a", options="header"]
+|===
+| Plugin
+| Description
+
+| None
+| Generates an IDEA module file. Also generates an IDEA project and workspace file if the project is the root project.
+
+| <<java_plugin,Java>>
+| Adds Java configuration to the module and project files.
+|===
+
+One focus of the IDEA plugin is to be open to customization. The plugin provides a standardized set of hooks for adding and removing content from the generated files.
+
+
+[[sec:idea_usage]]
+=== Usage
+
+To use the IDEA plugin, include this in your build script:
+
+++++
+<sample id="useIdeaPlugin" dir="idea" title="Using the IDEA plugin">
+            <sourcefile file="build.gradle" snippet="use-plugin"/>
+        </sample>
+++++
+
+The IDEA plugin adds a number of tasks to your project. The main tasks that you will use are the `idea` and `cleanIdea` tasks.
+
+[[sec:idea_tasks]]
+=== Tasks
+
+The IDEA plugin adds the tasks shown below to a project. Notice that the `clean` task does not depend on the `cleanIdeaWorkspace` task. This is because the workspace typically contains a lot of user specific temporary data and it is not desirable to manipulate it outside IDEA.
+
+[[ideatasks]]
+.IDEA plugin - Tasks
+[cols="a,a,a,a", options="header"]
+|===
+| Task name
+| Depends on
+| Type
+| Description
+
+| `idea`
+| `ideaProject`, `ideaModule`, `ideaWorkspace`
+| `-`
```

Comment:
I know it was in the original document but it's weird that the dash is a literal. IMHO it's not a reference to a task name and therefore shouldn't be a literal.